src/Security/src/Authentication.CloudFoundryBase/CloudFoundryTokenValidator.cs

```diff
@@ -51,6 +46,11 @@ public virtual bool ValidateAudience(IEnumerable<string> audiences, SecurityToke
                 return true;
             }
 
+            if (validationParameters != null && (audience.Equals(validationParameters.ValidAudience) || validationParameters.ValidAudiences?.Contains(audience) == true))
```

Comment:
```suggestion
            if (validationParameters != null && (audience == validationParameters.ValidAudience || validationParameters.ValidAudiences?.Contains(audience) == true))
```

<img width="1151" height="586" alt="image" src="https://github.com/user-attachments/assets/ad0a757a-7ba9-468e-9c02-8ae09e217b5d" />

Reply:
I know, that's why I suggested to use `==`, which avoids the entire issue.